pkg/controller/dmaas_backup_controller.go

```diff
@@ -196,3 +213,46 @@ func shouldProcessDMaaSBackup(dbkp v1alpha1.DMaaSBackup) (shouldProcess bool, ms
 	shouldProcess = true
 	return
 }
+
+// initializeDMaaSBackupMetaAndStatus update the dmaasbackup metadata and status
+// with required value
+func initializeDMaaSBackupMetaAndStatus(dbkp *v1alpha1.DMaaSBackup) {
+	dbkp.Status.Reason = ""
+
+	if dbkp.Status.Phase == v1alpha1.DMaaSBackupPhaseInProgress {
+		// We already initialized this dmaasbackup
+		return
+	}
+
+	// add dmaas operator finalizer
+	dbkp.ObjectMeta = addDMaaSFinalizer(dbkp.ObjectMeta)
+
+	// set dmaasbackup phase InProgress
+	dbkp.Status.Phase = v1alpha1.DMaaSBackupPhaseInProgress
+}
+
+// addDMaaSFinalizer add dmaas operator related finalizer to given object
+func addDMaaSFinalizer(obj metav1.ObjectMeta) metav1.ObjectMeta {
+	finalizers := obj.GetFinalizers()
+	for _, finalizer := range finalizers {
+		if finalizer == v1alpha1.DMaaSKey {
```

Comment:
i think we can avoid that since it will be added under finalizer, so keeping "finalizer" in the finalizer won't be a good idea. I'll change DMaaSKey to DMaaSFinalizer.

Reply:
ok